docs/our-approach-to-data.md

```diff
@@ -0,0 +1,267 @@
+
+An action object should contain a `type` key describing the action. All action types are defined in [`state/action-types.js`](https://github.com/Automattic/wp-calypso/blob/master/client/state/action-types.js). For example, to describe the intent of changing the state to include a few new post objects, you might create an action with the type `POSTS_RECEIVE`. Any other relevant properties can be included in this object if they are needed by the reducer function handler.
+
+As mentioned above, new actions should be added to `action-types.js`. Action types are considered global such that any state subtree's reducer can react to any action types dispatched through the system. The file should remain alphabetized, and we recommend suffixing the verb so that all actions within the same domain scope are in relative proximity. For example, rather than naming your actions `FETCH_POSTS` AND `RECEIVE_POSTS`, you should name them `POSTS_FETCH` AND `POSTS_RECEIVE`.
+
+### Data Normalization
+
+Because a Redux store is the [single source of truth](http://rackt.org/redux/docs/introduction/ThreePrinciples.html#single-source-of-truth) for the entire application state, it is important that all known data be tracked within the state tree and that it be well-structured. In your reducer functions, consider the data being manipulated in the tree and ensure that subjects are appropriately separated to minimize redundancy and to avoid synchronization concerns. When a subject needs to refer to another part of the tree, store a reference (likely an ID). Tracking an indexed set of items makes it easy to navigate the tree when needing to perform a lookup.
+
+As an example, consider that there are many variations of a "user" in the application. A user may be the current user, a subscriber to a site, or someone who has left a comment on a story in your Reader feed. Each of these display user data in different ways, and in some cases retrieve the data from different sources. However, they can all be classified as a user, and relations between the user and a display context can be established through references.
+
+The following state tree demonstrates how users, sites, and posts may be interrelated, but where data is normalized in a way such that it is always kept in sync, avoiding duplication, and facilitating lookup.
+
+```json
+{
+	"users": {
+		"items": {
+			"73705554": {
+				"ID": 73705554,
+				"login": "testonesite2014"
+			}
+		}
+	},
+	"sites": {
+		"items": {
+			"2916284": {
+				"ID": 2916284,
+				"name": "WordPress.com Example Blog",
+				"description": "Just another WordPress.com weblog",
+			}
+		},
+		"siteUsers": {
+			"2916284": {
+				"73705554": {
+					"roles": [ "administrator" ]
+				}
+			}
+		}
+	},
+	"posts": {
+		"items": {
+			"34": {
+				"ID": 34,
+				"title": "Hello World!",
+				"site_ID": 2916284,
+				"author_ID": 73705554
+			}
+		}
+	}
+}
+```
+
+### Container Components
```

Comment:
I don't consider `Query` components Container Components in the way we have fetching components wrapping UI/app components and I think it would be a bit confusing to have so many concepts around them.